Core.lua

```diff
@@ -381,7 +381,7 @@ do
 			end
 		else
 			local direction = column.sort or column.defaultsort or "asc";
-			if direction:lower() == "asc" then
```

Comment:
Hmm...   I don't want to break people...  but it's true the amount of people using non-lower-case is probably very slim.

I'd have no resistance to this whatsoever if you scanned the provided tables during create and forces the column sort order to lowercase.

Reply:
alternatively...   you could just use a switch statement and have cases for all 3 permutations of asc.  It's only 3 letters :)

Reply:
I've been away from lua too long... no switch statement.
However it's not a huge amount of comparisons either...  *shrug*

Reply:
Don't worry, I've covered that scenario. ;-) Here are the 3 commits in this pull request. Their commit messages explain the changes:

- Fix IsRowVisible() algorithm: https://github.com/ddumont/lib-st/pull/1/commits/25014c5fe87c86d7368e80a4de7ebc5c4e902100
- Improve table sorting performance: https://github.com/ddumont/lib-st/pull/1/commits/e93347a04d91c75e1bd771ef2c308d9e21d7a35c
- Rewrite sort to use constants, fixed asc-desc bug: https://github.com/ddumont/lib-st/pull/1/commits/e879fb7ea838362853b8139bc748cc505b4fd242

All changes are backwards-compatible drop-in fixes, mergable with no downsides. :-)